Client/Protocols/NavigationController.swift

```diff
@@ -20,4 +21,8 @@ extension UINavigationController: NavigationController {
     var fromViewController: UIViewController? {
         return transitionCoordinator?.viewController(forKey: .from)
     }
+
+    var topPresentedViewController: UIViewController? {
+        return presentedViewController
```

Comment:
If you name the variable presentedViewController, it will automatically point to the instance of this on UIViewController. 
I think this is better because I'm not a fan of naming this with the word "top", it can lead to confusion later on with regards to what it represents. For example if you have a ViewController A presenting a ViewController B, and B presents another ViewController C, if you ask A for it's topPresentedViewController, it would return B, but since it's named top you might expect it to return C. I don't think we have any use cases today where we have modals on top of modals, but we might in future.